app/src/main/java/com/github/multimatum_team/multimatum/MainActivity.kt

```diff
@@ -20,6 +23,23 @@ class MainActivity : AppCompatActivity() {
         startActivity(intent)
     }
 
+    fun openCodeScanner(view: View) {
```

Comment:
If I understand correctly, if permission is denied the function asks for the permission and then returns. Wouldn't it be better to also start the QR code reader once permission has been granted (i.e. move lines 38-39 out of the `else` statement)?